src/Interpreters/InterpreterExplainQuery.cpp

```diff
@@ -423,69 +489,45 @@ QueryPipeline InterpreterExplainQuery::executeImpl()
         {
             auto settings = checkAndGetSettings<QuerySyntaxSettings>(ast.getSettings());
 
-            ExplainAnalyzedSyntaxVisitor::Data data(getContext());
+            ContextPtr query_context = getContext();
+            if (query_context->getSettingsRef()[Setting::allow_experimental_analyzer])
+            {
+                bool explain_ok = explainQueryTree(ast.getExplainedQuery(), query_context, QueryTreeSettings{
+                    .run_passes = settings.run_query_tree_passes,
+                    .dump_tree = false,
+                    .dump_passes = false,
+                    .dump_ast = true,
+                    .passes = settings.query_tree_passes,
+                    .ast_one_line = settings.oneline,
+                }, buf);
+
+                if (explain_ok)
+                    break;
+                auto query_context_mutable = Context::createCopy(query_context);
+                query_context_mutable->setSetting("allow_experimental_analyzer", false);
```

Comment:
Looks strange

Reply:
Explain query tree that we reuse for explain syntax do not support queries other than `SELECT`. So for other types of queries current behavior with using old analyzer is kept.